python/tritonserver/_api/_tensor.py

```diff
@@ -509,16 +509,32 @@ def to_device(self, device: DeviceOrMemoryType) -> Tensor:
         if self.memory_type == MemoryType.CPU_PINNED and memory_type == MemoryType.CPU:
             return self
         if cupy is not None:
+            # DLPack does not support bytes type.
+            original_data_type = self.data_type
+            original_shape = self.shape
+            if self.data_type == DataType.BYTES:
```

Comment:
question - would this remain after we enable device memory on c++ side?

or is temporary - if temporary - we can add a note on when it can be removed?

Reply:
I think this can be permanent, because we (may?) want to support moving bytes tensors between host and device(s), which this adds the functionality.

Reply:
would like to see some examples - to understand a little - we also have the to_bytes_array / from_bytes_array - maybe those should also take a target device? currently assume to be numpy ... but not a blocker now

Reply:
I think the to_bytes_array underneath does the same hack and then call _to_numpy_on_host which underneath calls from_dlpack. Probably it is a good idea to unify the logic in the future, filed a ticket DLIS-7936 to keep track of this issue.